nft_ingester/src/index_synchronizer.rs

```diff
@@ -68,39 +69,37 @@ where
 
     pub async fn nft_run(
         &self,
-        rx: &tokio::sync::broadcast::Receiver<()>,
+        cancellation_token: CancellationToken,
         run_full_sync_threshold: i64,
-        timeout_duration: tokio::time::Duration,
     ) {
-        while rx.is_empty() {
-            if let Err(e) = self.synchronize_nft_asset_indexes(rx, run_full_sync_threshold).await {
+        while !cancellation_token.is_cancelled() {
```

Comment:
looks like synchronize\_nft\_asset\_indexes already accepts the token, I do not see why we need `while` here. `select` may be more suitable here, or even just `loop` with `break` on `error` return from the function

Reply:
each part in the subsequent call uses loops. the outer usage of cancellation token means that no new iterations will be started on this level, while the cancellation token in child functions acts to break inner loops